src/event/reader.rs

```diff
@@ -382,41 +394,41 @@ impl EventReader {
     /// Note: it may return an error indicating that the reader has already been removed. This means
     /// that another thread removes this reader from the ReaderGroup probably due to the host of this reader
     /// is assumed dead.
-    pub async fn reader_offline(&mut self) -> Result<(), EventReaderError> {
-        if !self.meta.reader_offline && self.rg_state.lock().await.check_online(&self.id).await {
-            info!("Putting reader {:?} offline", self.id);
+    async fn reader_offline_internal(
+        reader_id: Reader,
+        rg_state: Arc<Mutex<ReaderGroupState>>,
+        meta: &mut ReaderState,
+    ) -> Result<(), EventReaderError> {
+        let mut rg_guard = rg_state.lock().await;
+        let online = rg_guard.check_online(&reader_id).await;
+        drop(rg_guard);
+        if !meta.reader_offline && online {
```

Comment:
I believe this can all be inlined into one line. The if the guard is never assigned to a variable it will be dropped at the end of the line.